config-model-generator/src/main/java/io/strimzi/build/kafka/metadata/CommonConfigModelGenerator.java

```diff
@@ -0,0 +1,259 @@
+        String str = key.validator.toString();
+        try {
+            Pattern rangePattern = Pattern.compile("\\[([0-9.e+-]+|\\.\\.\\.),?([0-9.e+-]+|\\.\\.\\.)?,?([0-9.e+-]+|\\.\\.\\.)?\\]", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = rangePattern.matcher(str);
+            if (matcher.matches()) {
+                if (matcher.groupCount() == 3 && matcher.group(3) == null) {
+                    openRange(matcher, descriptor);
+                } else if (matcher.groupCount() == 3) {
+                    closedRange(matcher, descriptor);
+                } else if (matcher.groupCount() != 1) {
+                    throw new IllegalStateException(str);
+                }
+            } else {
+                throw new IllegalStateException(str);
+            }
+            return descriptor;
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Invalid range " + str + " for key " + key.name, e);
+        }
+    }
+
+    private void closedRange(Matcher matcher, ConfigModel descriptor) {
+        String maxStr = matcher.group(3);
+        if (maxStr.contains("e") || maxStr.contains(".") && !maxStr.contains("..")) {
+            descriptor.setMaximum(Double.parseDouble(maxStr));
+        } else if (!"...".equals(maxStr)) {
+            descriptor.setMaximum(Long.parseLong(maxStr));
+        }
+
+        String midStr = matcher.group(2);
+        if (!"...".equals(midStr)) {
+            throw new IllegalStateException();
+        }
+
+        String minStr = matcher.group(1);
+        if (minStr.contains("e") || minStr.contains(".") && !minStr.contains("..")) {
+            descriptor.setMinimum(Double.parseDouble(minStr));
+        } else if (!"...".equals(minStr)) {
+            descriptor.setMinimum(Long.parseLong(minStr));
+        }
+    }
+
+    private void openRange(Matcher matcher, ConfigModel descriptor) {
+        String maxStr = matcher.group(2);
+        if (maxStr.contains("e") || maxStr.contains(".") && !maxStr.contains("..")) {
+            descriptor.setMaximum(Double.parseDouble(maxStr));
+        } else if (!"...".equals(maxStr)) {
+            descriptor.setMaximum(Long.parseLong(maxStr));
+        }
+
+        String minStr = matcher.group(1);
+        if (minStr.contains("e") || minStr.contains(".") && !minStr.contains("..")) {
+            descriptor.setMinimum(Double.parseDouble(minStr));
+        } else if (!"...".equals(minStr)) {
+            descriptor.setMinimum(Long.parseLong(minStr));
+        }
+    }
+
+    protected Method methodFromReflection(Class<ConfigDef> origin, String methodName, Class... paramClasses) throws NoSuchMethodException {
+        Method method = origin.getDeclaredMethod(methodName, paramClasses);
+        doPrivileged((PrivilegedAction<ConfigDef>) () -> {
+            method.setAccessible(true);
+            return null;
+        });
+        return method;
+    }
+
+    protected abstract ConfigDef configsFromReflectivity();
```

Comment:
Think about the method naming: Does this class know how subclasses are obtaining the `ConfigDef`? Probably not, so it probably shouldn't mention the reflection implementation detail in the method name. Calling it `configDefs` would be enough.